ci/pkl-pipelines/pay-deploy/pay-cli.pkl

```diff
@@ -0,0 +1,45 @@
+amends "package://pkg.pkl-lang.org/github.com/alphagov/pkl-concourse-pipeline/pkl-concourse-pipeline@0.0.4#/Pipeline.pkl"
+
+import "../common/pipeline_self_update.pkl"
+import "../common/shared_resources.pkl"
+import "../common/PayResources.pkl"
+
+resources = new {
+  pipeline_self_update.PayPipelineSelfUpdateResource("pay-deploy/pay-cli.pkl", "master")
+  shared_resources.payCiGitHubResource
+  new PayResources.PayGitHubResource {
+    name = "pay-cli-git-main"
+    repoName = "pay-cli"
```

Comment:
I think this is more confusing than just ammending the source. You could instead do:

```
resources = new {
  pipeline_self_update.PayPipelineSelfUpdateResource("pay-deploy/pay-cli.pkl", "master")
  shared_resources.payCiGitHubResource
  new PayResources.PayGitHubResource {
    name = "pay-cli-git-main"
    source {
      branch = "main"
      commitFilter = new {
        exclude = new {"\\[automated release\\]"}
      }
    }
  }
```

It makes sense for the repo name because it's interpolated into a much longer thing, but for the others it feels confusing, especially since editors will now allow you to complete both branchName and source { branch } and it's not clear who wins (it's the ammendment of `source { branch` I think, but I'm not 100% :)

Reply:
👍 Yep, that's one of the many options that if we'd had time I wanted to talk about prior to raising this and I flip-flopped between the two myself 😄 . I'm happy with either approach, though I must slightly favour the as-is I suppose  (since that made it to the PR 😄 or perhaps I enjoyed experimenting with `pkl` and I've fallen foul of trying to be too clever in something I'm too new with 🤔 ).

Thanks for expressing your opinion and since you've raised this comment you must feel strongly enough about it, and I'm easy either way, so I'm happy to change it.